src/Model.js

```diff
@@ -509,8 +513,25 @@ const Model = class Model {
             }
         }
 
-        this._initFields(Object.assign({}, this._fields, mergeObj));
-        this._refreshMany2Many(m2mRelations); // eslint-disable-line no-underscore-dangle
+        const mergedFields = {
+            ...this._fields,
+            ...mergeObj,
+        };
+
+        const updatedModel = new ThisModel(this._fields);
+        updatedModel._initFields(mergedFields); // eslint-disable-line no-underscore-dangle
+
+        // determine if model would have different related models after update
+        updatedModel._refreshMany2Many(m2mRelations); // eslint-disable-line no-underscore-dangle
+        const relationsEqual = Object.keys(m2mRelations).every(name =>
+            !arrayDiffActions(this[name], updatedModel[name])
+        );
+
+        // do not apply updates if relations and fields are equal
+        if (relationsEqual && this.equals(updatedModel)) return;
+
+        this._initFields(mergedFields);
+        this._refreshMany2Many(m2mRelations);
```

Comment:
Maybe we can skip this in *all* cases when `relationsEqual` is truthy, not only in the case where  the fields are also referentially equal (i.e. `this.equals(updatedModel)` return truthy).